yjit/src/virtualmem.rs

```diff
@@ -59,7 +59,7 @@ pub trait Allocator {
 /// We may later change this to wrap an u32.
 /// Note: there is no NULL constant for CodePtr. You should use Option<CodePtr> instead.
 #[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Debug)]
-#[repr(C)]
+#[repr(C, packed)]
```

Comment:
I'm curious how much this change to CodePtr alone affects things. This doesn't change the size but allows the pointer to be stored unaligned: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=6f40323dcc2d2898fb01d2309d734102

Maybe the compiler sees this and gives `struct Block` a more compact layout?

Reply:
It has a relatively big impact for some reason.

## With CodePtr not packed
```
code_region_size:         9715712
yjit_alloc_size:         19684821
```

## With CodePtr packed
```
code_region_size:         9715712
yjit_alloc_size:         18179277
```